File: Swift/프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.swift
import Foundation

func solution(_ n:Int) -> [Int] {
    
    // 기본 골격 만들기
    var answer = [[Int]]()
    for i in 0..<n {
        answer.append(Array(repeating: 0, count: i+1))
    }
    
    var x = -1
    var y = 0
    var num = 1
    
    for i in 0..<n {
        for j in i..<n {
            if i % 3 == 0 {
                x += 1 // 아래로
            } else if i % 3 == 1 {
                y += 1 // 오른쪽으로
            } else {
                // 위쪽으로
                x -= 1
                y -= 1
            }
            answer[x][y] = num
            num += 1
        }
    }
    
    var result = [Int]()
    for ans in answer {
        for a in ans {
            result.append(a)
        }
    }
    
    return result
}

File: Python3/프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):  
    # 삼각형 만들기
    answer = []
    for i in range(n):
        answer.append([0] * (i + 1))
          
    x, y = -1, 0
    num = 1     
    for i in range(n): 
        print(i)    
        for j in range(i, n):
                        
            if i % 3 == 0: # 아래로              
                x += 1            
            elif i % 3 == 1: # 오른쪽으로               
                y += 1            
            else: # 위로             
                x -= 1                
                y -= 1            
                
            # 현재 위치에 숫자 채우기
            answer[x][y] = num            
            num += 1     
    return sum(answer, [])
